### ISSUE TITLE
Failing test: toque_zero_zero.js

### ISSUE THREAD

**Issue by Algunenano** [CartoDB/Windshaft#629]
The torque_zero_zero.js test is failing in my setup (Postgis 2.5 trunk, proj 5.1) where the point (0,0) is contained in all 4 tiles and not just the bottom ones:

```
$ ./node_modules/.bin/mocha -u tdd -t 5000 test/acceptance/torque_zero_zero.js


  torque tiles at 0,0 point
    1) tl
    2) tr
    ✓ bl
    ✓ br


  2 passing (63ms)
  2 failing

  1) torque tiles at 0,0 point tl:

      Uncaught AssertionError [ERR_ASSERTION]: [ anonymous {
    x__uint8: 1,
    y__uint8: 0,
    vals__uint8: [ 1 ],
    dates__uint16: [ 0 ] } ] deepEqual []
      + expected - actual

      -[
      -  {
      -    "dates__uint16": [
      -      0
      -    ]
      -    "vals__uint8": [
      -      1
      -    ]
      -    "x__uint8": 1
      -    "y__uint8": 0
      -  }
      -]
      +[]
      
      at test/acceptance/torque_zero_zero.js:83:24
      at test/support/test_client.js:87:16
      at Function.<anonymous> (lib/windshaft/backends/tile.js:61:13)
      at next (node_modules/step/lib/step.js:51:23)
      at lib/windshaft/renderers/torque/renderer.js:171:17
      at ctor.PSQLAdaptor.ctor._handleResult (lib/windshaft/renderers/torque/psql_adaptor.js:14:9)
      at Result.<anonymous> (node_modules/cartodb-psql/lib/psql.js:302:24)
      at Result.Query.handleReadyForQuery (node_modules/cartodb-psql/node_modules/pg/lib/query.js:144:10)
      at Connection.<anonymous> (node_modules/cartodb-psql/node_modules/pg/lib/client.js:184:19)
      at Socket.<anonymous> (node_modules/cartodb-psql/node_modules/pg/lib/connection.js:135:12)
      at addChunk (_stream_readable.js:274:12)
      at readableAddChunk (_stream_readable.js:261:11)
      at Socket.Readable.push (_stream_readable.js:218:10)
      at TCP.onread (net.js:581:20)

  2) torque tiles at 0,0 point tr:

      Uncaught AssertionError [ERR_ASSERTION]: [ anonymous {
    x__uint8: 0,
    y__uint8: 0,
    vals__uint8: [ 1 ],
    dates__uint16: [ 0 ] } ] deepEqual []
      + expected - actual

      -[
      -  {
      -    "dates__uint16": [
      -      0
      -    ]
      -    "vals__uint8": [
      -      1
      -    ]
      -    "x__uint8": 0
      -    "y__uint8": 0
      -  }
      -]
      +[]
      
      at test/acceptance/torque_zero_zero.js:83:24
      at test/support/test_client.js:87:16
      at Function.<anonymous> (lib/windshaft/backends/tile.js:61:13)
      at next (node_modules/step/lib/step.js:51:23)
      at lib/windshaft/renderers/torque/renderer.js:171:17
      at ctor.PSQLAdaptor.ctor._handleResult (lib/windshaft/renderers/torque/psql_adaptor.js:14:9)
      at Result.<anonymous> (node_modules/cartodb-psql/lib/psql.js:302:24)
      at Result.Query.handleReadyForQuery (node_modules/cartodb-psql/node_modules/pg/lib/query.js:144:10)
      at Connection.<anonymous> (node_modules/cartodb-psql/node_modules/pg/lib/client.js:184:19)
      at Socket.<anonymous> (node_modules/cartodb-psql/node_modules/pg/lib/connection.js:135:12)
      at addChunk (_stream_readable.js:274:12)
      at readableAddChunk (_stream_readable.js:261:11)
      at Socket.Readable.push (_stream_readable.js:218:10)
      at TCP.onread (net.js:581:20)
```

This can be checked with the following query generated by torque:
```sql
WITH par AS (WITH innerpar AS (SELECT 1.0/((19567.87939453125)*128) as resinv, ST_MakeEnvelope(-5009377.125, 5009377.125, 0, 0, 3857) as b_ext, ST_MakeEnvelope(-5009377.125, 5009377.125, 0, 0, 3857) as ext) SELECT (19567.87939453125)*128 as res, innerpar.resinv as resinv, innerpar.b_ext as b_ext, st_xmin(innerpar.ext) as xmin, st_ymin(innerpar.ext) as ymin, round((st_xmax(innerpar.ext) - st_xmin(innerpar.ext))*innerpar.resinv) - 1 as maxx, round((st_ymax(innerpar.ext) - st_ymin(innerpar.ext))*innerpar.resinv) - 1 as maxy FROM innerpar) SELECT xx x__uint8, yy y__uint8, array_agg(c) vals__uint8, array_agg(d) dates__uint16 FROM ( select GREATEST(0 - 0, LEAST(p.maxx + 0, round((st_x(i.the_geom_webmercator) - p.xmin)*resinv))) as xx, GREATEST(0 - 0, LEAST(p.maxy + 0, round((st_y(i.the_geom_webmercator) -
 p.ymin)*resinv))) as yy , count(cartodb_id) c , floor((cartodb_id - 1)/1) d FROM (select 1 cartodb_id, ST_Transform(ST_SetSRID(ST_MakePoint(0, 0), 4326), 3857) the_geom_webmercator) i, par p WHERE i.the_geom_webmercator && p.b_ext GROUP BY xx, yy, d ) cte, par GROUP BY x__uint8, y__uint8
```

The result with postgis 2.5, proj 5.1:
 ```
 x__uint8 | y__uint8 | vals__uint8 | dates__uint16 
----------+----------+-------------+---------------
        1 |        0 | {1}         | {0}
(1 row)
```

The result with Postgis 2.2.1 and Proj 4.9.2:
```
 x__uint8 | y__uint8 | vals__uint8 | dates__uint16 
----------+----------+-------------+---------------
(0 rows)
```

I think that this lack of precision was in fact solved by @jgoizueta for Proj 5.1 in https://github.com/OSGeo/proj.4/pull/928 so now 0,0 is still located at exact 0,0 after the transformation, which is contained by all 4 tiles.

**Comment by jgoizueta** [comment]
We were testing the PROJ 0,0 webmercator bug (or lack of accuracy) actually with this test.
Strictly we should modify the test to accept the correct result (all four tiles contain the point), but then we also should accept the current result as valid as we have been doing, as long as we'll keep using the old PROJ versions for testing.

For MVT tiles I'd change the test to use a `buffersize` of 1, giving some tolerance so that the point is the four tiles even with the inprecise PROJ. Is there something equivalent to `buffersize` in torque?

**Comment by Algunenano** [comment]
I've reworked the PR to keep the transformation and accept as valid the output with the bug fixed (proj 5.1).
